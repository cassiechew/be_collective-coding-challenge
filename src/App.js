import { Header } from './components/Header';
import { Footer } from './components/Footer';
import { FileView } from './containers/Files';

import { useState } from 'react';
import './App.scss';

function App() {
  const [numFiles, setNumFiles] = useState(0);
  const [fileSize, setFileSize] = useState(0);

  return (
    <div className="container__app">
      <Header className="container__header" title="File Viewer" />
      <FileView setParentNumFiles={setNumFiles} setParentFileSize={setFileSize} />
      <Footer numFiles={numFiles} fileSize={fileSize} />

    </div>
  );
}

export default App;
