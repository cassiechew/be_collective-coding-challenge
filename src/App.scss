div {
  .container__app {
    height: 100vh;
  }

  .root {
    height: 100vh;
  }
}

