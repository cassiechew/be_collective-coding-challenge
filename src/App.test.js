import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import App from './App';
import React from 'react';

test('renders file viewer text', () => {
  render(<App />);
  const linkElement = screen.getByText(/File Viewer/i);
  expect(linkElement).toBeInTheDocument();
});

describe('snapshot test', () => {
  test('testing app', () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});