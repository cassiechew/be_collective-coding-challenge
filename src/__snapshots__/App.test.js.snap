// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test testing app 1`] = `
<div
  className="container__app"
>
  <div
    className="container__header"
  >
    <h1>
      File Viewer
    </h1>
  </div>
  <div
    className="container"
  >
    <div
      className="container__file-list"
    />
  </div>
  <div
    className="container__footer"
  >
    <h3>
      Total Files: 
      0
    </h3>
    <h3>
      Total File Size: 
      0
    </h3>
  </div>
</div>
`;
